helios-testing/src/main/java/com/spotify/helios/testing/DockerHost.java

```diff
@@ -0,0 +1,167 @@
+package com.spotify.helios.testing;
+
+import com.google.common.net.HostAndPort;
+
+import java.net.URI;
+
+import static com.google.common.base.Optional.fromNullable;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.lang.System.getenv;
+
+/**
+ * Represents a dockerd endpoint. A codified DOCKER_HOST.
+ */
+public class DockerHost {
+
+  public static final int DEFAULT_PORT = 2375;
+  public static final String DEFAULT_HOST = "localhost";
+  public static final String DEFAULT_UNIX_ENDPOINT = "unix:///var/run/docker.sock";
+
+  private final String host;
+  private final URI uri;
+  private final URI bindURI;
+  private final String address;
+  private final int port;
+  private final String dockerCertPath;
+
+  private DockerHost(final String endpoint, final String dockerCertPath) {
+    if (endpoint.startsWith("unix://")) {
+      this.port = 0;
+      this.address = DEFAULT_HOST;
+      this.host = endpoint;
+      this.uri = URI.create(endpoint);
+      this.bindURI = URI.create(endpoint);
+    } else {
+      final String stripped = endpoint.replaceAll(".*://", "");
+      final HostAndPort hostAndPort = HostAndPort.fromString(stripped);
+      final String hostText = hostAndPort.getHostText();
+      final String scheme = isNullOrEmpty(dockerCertPath) ? "http" : "https";
+
+      this.port = hostAndPort.getPortOrDefault(defaultPort());
+      this.address = isNullOrEmpty(hostText) ? DEFAULT_HOST : hostText;
+      this.host = address + ":" + port;
+      this.uri = URI.create(scheme + "://" + address + ":" + port);
+      this.bindURI = URI.create("tcp://" + address + ":" + port);
+    }
+
+    this.dockerCertPath = dockerCertPath;
+  }
+
+  /**
+   * Get a docker endpoint usable for instantiating a new DockerHost with DockerHost.from(endpoint).
+   *
+   * @return The hostname.
+   */
+  public String host() {
+    return host;
+  }
+
+  /**
+   * Get the docker rest uri.
+   *
+   * @return The uri of the host.
+   */
+  public URI uri() {
+    return uri;
+  }
+
+  /**
+   * Get the docker rest bind uri.
+   *
+   * @return The uri of the host for binding ports (or setting $DOCKER_HOST).
+   */
+  public URI bindURI() {
+    return bindURI;
+  }
+  /**
+   * Get the docker endpoint port.
+   *
+   * @return The port.
+   */
+  public int port() {
+    return port;
+  }
+
+  /**
+   * Get the docker ip address or hostname.
+   *
+   * @return The ip address or hostname.
+   */
+  public String address() {
+    return address;
+  }
+
+  /**
+   * Get the path to certificate and key for connecting to Docker via HTTPS.
+   *
+   * @return The path to the certificate.
+   */
+  public String dockerCertPath() {
+    return dockerCertPath;
+  }
+
+  /**
+   * Create a {@link DockerHost} from DOCKER_HOST and DOCKER_PORT env vars.
+   *
+   * @return The DockerHost object.
+   */
+  public static DockerHost fromEnv() {
+    String defaultEndpoint;
```

Comment:
It was like that when I copy pasta'd it from services-common. I'll add a TODO for now and fix it when we copy the implementation into the docker-client in a future PR.